radicle-httpd/src/api/v1/sessions.rs

```diff
@@ -1,131 +1,95 @@
-use std::collections::HashMap;
-use std::convert::TryInto;
-use std::env;
 use std::iter::repeat_with;
-use std::str::FromStr;
 
 use axum::extract::State;
 use axum::response::IntoResponse;
-use axum::routing::{get, post};
+use axum::routing::{post, put};
 use axum::{Json, Router};
-use ethers_core::utils::hex;
-use hyper::http::uri::Authority;
+use radicle::crypto::{PublicKey, Signature};
+use serde::{Deserialize, Serialize};
 use serde_json::json;
-use siwe::Message;
 use time::{Duration, OffsetDateTime};
 
-use crate::api::auth::{AuthRequest, AuthState, DateTime, Session};
+use crate::api::auth::{AuthState, DateTime, Session};
 use crate::api::axum_extra::Path;
 use crate::api::error::Error;
 use crate::api::Context;
 
 pub const UNAUTHORIZED_SESSIONS_EXPIRATION: Duration = Duration::seconds(60);
+pub const AUTHORIZED_SESSIONS_EXPIRATION: Duration = Duration::weeks(1);
 
 pub fn router(ctx: Context) -> Router {
     Router::new()
         .route("/sessions", post(session_create_handler))
-        .route(
-            "/sessions/:id",
-            get(session_get_handler).put(session_signin_handler),
-        )
+        .route("/sessions/:id", put(session_signin_handler))
         .with_state(ctx)
 }
 
+#[derive(Debug, Deserialize, Serialize)]
+struct AuthChallenge {
+    sig: Signature,
+    pk: PublicKey,
+}
+
 /// Create session.
 /// `POST /sessions`
 async fn session_create_handler(State(ctx): State<Context>) -> impl IntoResponse {
+    let rng = fastrand::Rng::new();
+    let session_id = repeat_with(|| rng.alphanumeric())
+        .take(32)
+        .collect::<String>();
+    let signer = ctx.profile.signer().map_err(Error::from)?;
     let expiration_time = OffsetDateTime::now_utc()
         .checked_add(UNAUTHORIZED_SESSIONS_EXPIRATION)
         .unwrap();
+    let auth_state = AuthState::Unauthorized {
+        public_key: *signer.public_key(),
+        expires_at: DateTime(expiration_time),
+    };
     let mut sessions = ctx.sessions.write().await;
-    let (session_id, nonce) = create_session(&mut sessions, DateTime(expiration_time));
-
-    Json(json!({ "id": session_id, "nonce": nonce }))
-}
+    sessions.insert(session_id.clone(), auth_state);
 
-/// Get session.
-/// `GET /sessions/:id`
-async fn session_get_handler(
-    State(ctx): State<Context>,
-    Path(id): Path<String>,
-) -> impl IntoResponse {
-    let sessions = ctx.sessions.read().await;
-    let session = sessions.get(&id).ok_or(Error::NotFound)?;
-
-    match session {
-        AuthState::Authorized(session) => {
-            Ok::<_, Error>(Json(json!({ "id": id, "session": session })))
-        }
-        AuthState::Unauthorized {
-            nonce,
-            expiration_time,
-        } => Ok::<_, Error>(Json(
-            json!({ "id": id, "nonce": nonce, "expirationTime": expiration_time }),
-        )),
-    }
+    Ok::<_, Error>(Json(
+        json!({"session_id": session_id, "public_key": signer.public_key()}),
```

Comment:
```suggestion
        json!({"session_id": sessionId, "publicKey": signer.public_key()}),
```

Reply:
Fixed in ca25ea9ef95615df774d65fe9de31df899a240d7